dango/testing/tests/warp.rs

```diff
@@ -306,3 +315,315 @@ fn receive_minting_synth() {
+                eth_usdc_denom.clone() => BalanceChange::Increased(100),
+                sol_usdc_denom.clone() => BalanceChange::Increased(200),
+                alloyed_usdc_denom.clone() => BalanceChange::Increased(125),
+            });
+
+        suite
+            .balances()
+            .should_change(contracts.warp.address(), btree_map! {
+                eth_usdc_denom.clone() => BalanceChange::Increased(50),
+                sol_usdc_denom.clone() => BalanceChange::Increased(75),
+            });
+    }
+
+    // Recap the balances of user1.
+    // eth_usdc => 100
+    // sol_usdc => 200
+    // alloyed_usdc => 125 | 50 from eth, 75 from sol
+
+    // Send 20 alloyed_usdc to eth.
+    {
+        suite.balances().refresh_all();
+
+        // Get the current merkle tree.
+        let mut tree = suite
+            .query_wasm_smart(contracts.hyperlane.merkle, merkle::QueryTreeRequest {})
+            .should_succeed();
+
+        let recipient: Addr32 = Addr::mock(2).into();
+
+        // Send 20 alloyed to eth.
+        suite
+            .hyperlane()
+            .send_transfer(
+                &mut accounts.user1,
+                eth_domain,
+                recipient,
+                Coin::new(alloyed_usdc_denom.clone(), 20).unwrap(),
+            )
+            .should_succeed();
+
+        // Insert the message into the tree.
+        let token_msg = TokenMessage {
+            recipient,
+            amount: Uint128::new(20),
+            metadata: Default::default(),
+        };
+
+        let msg = Message {
+            version: MAILBOX_VERSION,
+            nonce: 0,
+            origin_domain: MOCK_LOCAL_DOMAIN,
+            sender: contracts.warp.into(),
+            destination_domain: eth_domain,
+            recipient: eth_usdc_recipient,
+            body: token_msg.encode(),
+        };
+
+        tree.insert(msg.encode().keccak256()).unwrap();
+
+        // Check if the merkle tree has been updated.
+        suite
+            .query_wasm_smart(contracts.hyperlane.merkle, merkle::QueryTreeRequest {})
+            .should_succeed_and_equal(tree);
+
+        // Verify balances.
+        suite
+            .balances()
+            .should_change(accounts.user1.address(), btree_map! {
+                alloyed_usdc_denom.clone() => BalanceChange::Decreased(20),
+            });
+
+        suite
+            .balances()
+            .should_change(contracts.warp.address(), btree_map! {
+                eth_usdc_denom.clone() => BalanceChange::Decreased(20),
+            });
+    }
+
+    // 20 alloyed_usdc has been sent via eth.
+    // Try sent 35 more. This should fail (30 left).
+    {
+        suite
+            .hyperlane()
+            .send_transfer(
+                &mut accounts.user1,
+                eth_domain,
+                Addr::mock(2).into(),
+                Coin::new(alloyed_usdc_denom.clone(), 35).unwrap(),
+            )
+            .should_fail_with_error("subtraction overflow: 30 - 35");
```

Comment:
Instead of comparing with a string, it's always better to compare with the error type, so that tests don't break if we change the error message:

```rust
.shoud_fail_with_error(MathError::overflow_sub::<Uint128>(30, 35));
```